### PR TITLE
build: add support for go 1.22

### PR DESCRIPTION
**Pull request by subhamkrai**
Few day's back go 1.22 is release. Let's add that
as valid version to build on.

https://tip.golang.org/doc/go1.22

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
Do we want to make the 1.22 default version in CI?

@travisn ^^

**Review by travisn — CHANGES_REQUESTED** [review]
With this PR, how about if we just update the build.yml action `linux-build-all` to include 1.22 in the go-version matrix? This way we know 1.22 can build it successfully. Then in a future PR we can update other CI actions to use 1.22 by default.

**Comment by subhamkrai** [comment]
> With this PR, how about if we just update the build.yml action `linux-build-all` to include 1.22 in the go-version matrix? This way we know 1.22 can build it successfully. Then in a future PR we can update other CI actions to use 1.22 by default.

done

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go-version: ["1.21"]
+        go-version: ["1.21","1.22"]
```
Comment (subhamkrai) [inline .github/workflows/build.yml:80]:
```suggestion
        go-version: ["1.21","1.22"]
```

linux build on go 1.22 is working on the system not sure why `build-all` is failing. Maybe we are too early to upgrade. Will wait for week or so to see if other dependencies updates before debugging this.

**Comment by travisn** [comment]
Let's see if Go 1.22.1 fixes this issue...
@Mergifyio rebase

**Comment by travisn** [comment]
Go 1.22.0 is being used instead of 1.22.1 after the rebase

File: .github/workflows/build.yml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/build.yml:74]: fixed

**Review by subhamkrai — COMMENTED** [review]
@travisn do you think we should make `go 1.22` default version? Also, I guess we can now revert the commits https://github.com/rook/rook/pull/13793/commits as descriptions are more organized?

**Review by travisn — APPROVED** [review]
> @travisn do you think we should make `go 1.22` default version? Also, I guess we can now revert the commits https://github.com/rook/rook/pull/13793/commits as descriptions are more organized?

For now let's merge this, then separately change the default to 1.22 and other updates.

**Comment by smoshiur1237** [comment]
Request to do a backport of this PR to release-1.13

We have received a request to fix vulnerability issues in rook 1.13 (built with go 1.20.x) which can be fixed by having support with golang 1.22.x in release 1.13 branch. I had discussion with @travisn and @parth-gr over chat. So once the support has been added in rook 1.13 and fix  vulnerability issue, our internal team will be allowed to do the upgrade to rook 1.14.

**Comment by travisn** [comment]
@smoshiur1237 We can backport this to allow building with go 1.22, but this PR won't change the version of go that release builds use, and we don't have another 1.13.x release planned currently. But if you anyway have your own build process, we can go ahead with this backport.

**Comment by smoshiur1237** [comment]
@travisn yes please, we can fetch the backport PR and have our own build process. Would you please trigger the backport or I need to do manual backport?

**Pull request merged**